internal/pkg/state/view.go

```diff
@@ -198,6 +198,24 @@ func (v *View) MarketEscrowBalance(ctx context.Context, addr addr.Address) (foun
 	return
 }
 
+// MarketGetDeal finds a deal by (resolved) provider and deal id
+func (v *View) MarketGetDealIDs(ctx context.Context, addr addr.Address) ([]abi.DealID, error) {
+	marketState, err := v.loadMarketActor(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	byParty := market.AsSetMultimap(StoreFromCbor(ctx, v.ipldStore), marketState.DealIDsByParty)
+	var providerDealIds []abi.DealID
+	if err = byParty.ForEach(addr, func(i abi.DealID) error {
+		providerDealIds = append(providerDealIds, i)
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+	return providerDealIds, err
+}
```

Comment:
Giant. I would prefer not to expose this so callers don't think it's reasonable. You could expose a ForEach instead, or just the particular behaviour of finding one deal that motivated this.

Reply:
It's MarketHasDealID now and does the traversal internally.